### PR TITLE
Refactor database changes

### PR DESCRIPTION
**Pull request by Chibbluffy**

### CODE REVIEW

File: src/main/java/dev/carrico/CardsSpringBackEnd/CardsSpringBackEndApplication.java [hunk 1]
```diff
@@ -6,10 +6,8 @@
 import org.springframework.context.annotation.ComponentScan;
 import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
 
-@ComponentScan(basePackages = {"dev.carrico"}) // Tells spring to check all packages for components
+@ComponentScan(basePackages = {"dev.carrico"})
 @EnableJpaRepositories(basePackages = {"dev.carrico.repos"})
-// will tell spring data where to find your entities
-// in base hibernate this was a mapping tag
```
Comment (Lcarrico) [inline src/main/java/dev/carrico/CardsSpringBackEnd/CardsSpringBackEndApplication.java:12]: good bye comments hahaahaa

File: src/main/java/dev/carrico/controllers/TagLinkController.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            tagLinks = this.tagLinkService.getTagLinksByTagId(Integer.parseInt(tagId));
+        }
+        else if (!cardId.isEmpty()){
+            tagLinks = this.tagLinkService.getTagLinksByCardId(Integer.parseInt(cardId));
+        }
+        else {
+            tagLinks = null;
+        }
+        return tagLinks;
+    }
```
Comment (Lcarrico) [inline src/main/java/dev/carrico/controllers/TagLinkController.java:46]: There's probs a better way to do this... lets keep this on our TODO to clean up a bit

File: src/main/java/dev/carrico/CardsSpringBackEnd/CardsSpringBackEndApplication.java (hunk 1 above)
Comment (Chibbluffy) [reply src/main/java/dev/carrico/CardsSpringBackEnd/CardsSpringBackEndApplication.java:12]: Lollll

File: src/main/java/dev/carrico/repos/StackLinkRepo.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Repository;
+
+import java.util.Set;
+
+@Component
+@Repository
+public interface StackLinkRepo extends CrudRepository<StackLink, Integer> {
+    Set<StackLink> findByLearnerId(int learnerId);
+    Set<StackLink> findByRelationship(String relationship);
```
Comment (Lcarrico) [inline src/main/java/dev/carrico/repos/StackLinkRepo.java:14]: This should be learnerId and relationship maybe?

File: src/test/java/dev/carrico/CardsSpringBackEnd/ServiceTests/LearnerServiceTests.java [hunk 4]
```diff
@@ -18,6 +20,9 @@
     @Autowired
     LearnerService ls;
 
+    @Autowired
+    StackService ss;
```
Comment (Lcarrico) [inline src/test/java/dev/carrico/CardsSpringBackEnd/ServiceTests/LearnerServiceTests.java:24]: @Autowired strikes again 🤣

**Review by Lcarrico — APPROVED** [review]
looks gooooooooooood

File: src/main/java/dev/carrico/repos/StackLinkRepo.java (hunk 3 above)
Comment (Chibbluffy) [reply src/main/java/dev/carrico/repos/StackLinkRepo.java:14]: 👍 Was gonna ask you guys what you thought. We might not need the relationship one at all

**Pull request merged**